File: static/js/koala_op/account.js
/**
 * Created by wangfeng on 2017/3/20.
 */
var account = function () {

    var SCHOOL_ID;
    var CLASS_JOIN_CODE;

    function get_city(province_id) {
        var csrf_token = getCookie('csrftoken');
        $.ajax({
            url: 'get_sub_area/',
            type: 'get',
            data: {
                csrfmiddlewaretoken: csrf_token,
                province_id: province_id
            },
            success: function (data) {
                data = JSON.parse(data);
                var city = [];
                for(var i = 0;i < data.length;i++){
                    city.push({
                        id: data[i].id,
                        text:data[i].regionName
                    });
                }
                $('#city').select2({
                    data: city,
                    language: 'zh-CN'
                });
                get_district($('#city').val());
            }
        })
    }

    function get_district(city_id) {
        var csrf_token = getCookie('csrftoken');
        $.ajax({
            url: 'get_sub_area/',
            type: 'get',
            data: {
                csrfmiddlewaretoken: csrf_token,
                city_id: city_id
            },
            success: function (data) {
                data = JSON.parse(data);
                var district = [];
                for(var i = 0;i < data.length;i++){
                    district.push({
                        id: data[i].regionCode,
                        text:data[i].regionName
                    });
                }
                $('#district').select2({
                    data: district,
                    language: 'zh-CN'
                });
                get_school($('#district').val());
            }
        })
    }

    function get_school(district_id) {
        var csrf_token = getCookie('csrftoken');
        $.ajax({
            url: 'get_sub_area/',
            type: 'get',
            data: {
                csrfmiddlewaretoken: csrf_token,
                district_id: district_id
            },
            success: function (data) {
                data = JSON.parse(data);
                var school = [];
                for(var i = 0;i < data.length;i++){
                    school.push({
                        id: data[i].id,
                        text:data[i].name
                    });
                }
                $('#school').select2({
                    data: school,
                    language: 'zh-CN'
                });
            }
        })
    }

    return {
        init: function () {
            $("#province").on('change',function(){

                $("#city").empty();
                $("#district").empty();
                $("#school").empty();

                get_city($("#province").val());

            });
            $("#city").on('change',function(){

                $("#district").empty();
                $("#school").empty();

                get_district($("#city").val());

            });
            $("#district").on('change',function(){

                $("#school").empty();

                get_school($("#district").val());

            });
            $("#school").on('change',function(){

                D($('#school').val());

            });
        },
        submit_file: function () {
            var csrf_token = getCookie('csrftoken');
            var file = $('#student_file')[0].files[0];
            // D(file);
            var form = new FormData();
            form.append('csrfmiddlewaretoken', csrf_token);
            form.append('file', file);
            D('upload');
            $.ajax({
                url: 'upload/',
                type: 'post',
                data: form,
                processData: false,
                contentType: false,
                success: function (data) {
                    $('#student_table').html(data);
                }
            })
        },
        create_school_master: function () {
            var school_id = $('#school').val();
            SCHOOL_ID = school_id;
            var csrf_token = getCookie('csrftoken');
            $.ajax({
                url: 'create_school_master/',
                type: 'post',
                data: {
                    csrfmiddlewaretoken: csrf_token,
                    school_id: school_id
                },
                success: function (data) {
                    data = JSON.parse(data);
                    D(data);
                    $('#school_master').html(data['account'] + ' 密码：' + data['psd'])
                }
            })
        },
        create_class: function () {
            // 创建班级老师，创建班级
            var school_id = $('#school').val();
            var grade = $('#grade').val();
            var name = $('#grade option[value=' + grade.toString() + ']').text() + $('#class').val();
            D(name);
            var csrf_token = getCookie('csrftoken');
            $.ajax({
                url: 'create_class/',
                type: 'post',
                data: {
                    csrfmiddlewaretoken: csrf_token,
                    school_id: school_id,
                    grade: grade,
                    name: name
                },
                success: function (data) {
                    data = JSON.parse(data);
                    D(data);
                    CLASS_JOIN_CODE = data.join_code;
                    $('#class_info').html(data.account + ' ' + data.psd + ' ' + data.join_code);
                }
            })
        },
        create_students: function () {
            var csrf_token = getCookie('csrftoken');
            // D($('#school').val());
            $.ajax({
                url: 'create_students/',
                type: 'post',
                data: {
                    csrfmiddlewaretoken: csrf_token,
                    school_id: $('#school').val(),
                    join_code: CLASS_JOIN_CODE
                },
                success: function (data) {
                    data = JSON.parse(data);
                    D(data);
                }
            })
        }
    }
}();

File: koala_op/views/account.py
# coding=utf-8
from datetime import datetime, timedelta
import json
import urllib
import urllib2

from django.shortcuts import render, render_to_response
from django.http import HttpResponse
from django.template import RequestContext
from django.views.decorators.csrf import csrf_protect, ensure_csrf_cookie

from koala_op.logic import account_controller as AC

rc = AC.RegionController()


@ensure_csrf_cookie
def index(request):
    global rc
    try:
        print request.session['file_path']
    except Exception as e:
        pass
    provinces = rc.get_top_region()
    cities = rc.get_sub_region(provinces[0]['id'])
    districts = rc.get_sub_region(cities[0]['id'])
    schools = rc.get_school(districts[0]['regionCode'])
    context = {
        'provinces': provinces,
        'cities': cities,
        'districts': districts,
        'schools': schools
    }
    return render(request, 'koala_op/account/account.html', context=context)


def upload(request):
    f = request.FILES['file']
    if 'csv' in f.name:
        data = []
        for i in f:
            line = i.strip('\r').split(',')
            print line[5] == '男'
            data.append(line)
        file_path = 'static/file/tmp_file/{}'.format(f.name)
        with open(file_path, 'w+') as f_out:
            for chunks in f.chunks():
                f_out.write(chunks)
        request.session['file_path'] = file_path
        return render(request, 'koala_op/account/student_table.html', {'data': data})
    else:
        return HttpResponse(status=400, content='file type error')


def get_sub_area(request):
    global rc
    if request.GET.get('province_id'):
        p_id = request.GET.get('province_id')
        res = rc.get_sub_region(p_id)
        return HttpResponse(json.dumps(res))
    elif request.GET.get('city_id'):
        c_id = request.GET.get('city_id')
        res = rc.get_sub_region(c_id)
        return HttpResponse(json.dumps(res))
    elif request.GET.get('district_id'):
        code = request.GET.get('district_id')
        res = rc.get_school(code)
        return HttpResponse(json.dumps(res))
    else:
        return HttpResponse(status=400, content='parameter not valid')


def create_school_master(request):
    """
    创建校长
    :param request:
    :return:
    """
    ac = AC.AccountController()
    school_id = request.POST.get('school_id')
    if school_id:
        account, psd = ac.create_school_master(school_id)
        return HttpResponse(json.dumps(dict(account=account, psd=psd)))


def create_class(request):
    ac = AC.AccountController()
    school_id = request.POST.get('school_id')
    grade = request.POST.get('grade')
    name = request.POST.get('name')
    if school_id:
        res = ac.create_class(school_id, dict(grade=grade, full_name=name))
        return HttpResponse(json.dumps(res))


def create_students(request):
    ac = AC.AccountController()
    school_id = request.POST.get('school_id')
    join_code = request.POST.get('join_code')
    file_path = request.session['file_path']
    print school_id, join_code, file_path
    if school_id:
        res = ac.create_student_from_file(school_id, join_code, file_path)
        return HttpResponse(json.dumps(res))

